server/src/service/item.rs

```diff
@@ -0,0 +1,124 @@
+// org.freedesktop.Secret.Item
+
+use std::collections::HashMap;
+
+use oo7::dbus::api::properties::Properties;
+use serde::{Deserialize, Serialize};
+use zbus::{dbus_interface, fdo, zvariant, ObjectServer};
+use zvariant::{ObjectPath, OwnedObjectPath};
+
+use crate::KEYRING;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Item {
+    locked: bool,
+    attributes: Properties,
+    created: u64,
+    modified: u64,
+    secret: Vec<u8>,
+    path: OwnedObjectPath,
+}
+
+#[dbus_interface(name = "org.freedesktop.Secret.Item")]
+impl Item {
+    pub async fn delete(
+        &self,
+        #[zbus(object_server)] object_server: &ObjectServer,
+    ) -> fdo::Result<ObjectPath> {
+        let mut properties: HashMap<&str, &str> = Default::default();
+
+        for (key, value) in self.attributes().await.attributes().unwrap().into_iter() {
```

Comment:
The attributes getter is already giving you just the attributes. This whole thing is not needed

Reply:
Yeah I agree. But, the attributes getter returns `&HashMap<String, String>` and I need `HashMap<&str, &str>`